tuna/go_fish.py

```diff
@@ -40,29 +41,56 @@ def parse_args():
   parser = argparse.ArgumentParser()
 
   parser.add_argument('lib',
+                      nargs='?',
+                      default=Library.MIOPEN,
                       type=Library,
                       help="Specify library to run",
                       choices=Library)
 
+  parser.add_argument('--yaml',
+                      dest='yaml',
+                      default=None,
+                      help='Path to yaml input file')
+
   args, _ = parser.parse_known_args()
   return vars(args)
 
 
 def main():
   """Main function to start Tuna"""
   LOGGER.info(sys.argv)
+  LOGGER.info(len(sys.argv))
   args = parse_args()
-  library = get_library(args)
+  if '--yaml' in sys.argv and len(sys.argv) > 4:
+    LOGGER.error('Command like arguments not accepted with yaml file')
+    return False
+
+  #case no yaml file
+  libraries = []
+  libraries.append(get_library(args))
+  yaml_files = [args['yaml']]
+
+  #case with yaml file
+  if args['yaml']:
+    yaml_files = parse_yaml(args['yaml'], args['lib'])
+    libraries = []
+    for _ in range(len(yaml_files)):
+      libraries.append(get_library(args))
```

Comment:
Maybe the list of libraries isn't necessary. It seems that they will all be the same since there is only 1 lib argument.

Reply:
if the yaml files gets split and it turns out there is only 1 tuning instruction there length of libraries is 1. If the initial yaml file has multiple tuning instructions and it gets split into 3 yaml files, then size of libraries will be 3, and each lib will get 1 of the 3 yaml files

Reply:
I mean the input library to go_fish can only ever be 1 library, so why make a list here? Couldn't we just use the 1 library and iterate over the yaml_files? I don't see why we're zipping a list of duplicates with the yaml_files.

Reply:
yes, ill update